src/visualisation/VTKWriter.cpp

```diff
@@ -83,6 +83,115 @@ void VTKWriter::write_analytical(const Solution &solution, const std::string &fi
     VTKWriter::vtk_prepare_and_write((filename + ending).c_str(), u, v, w, p, T);
 }
 
+/**
+ * Only for debugging
+ * @param field
+ * @param filename
+ * @param var_name
+ */
+void VTKWriter::write_field(const Field &field, const std::string &file_name, const std::string& var_name) {
+    std::string xml_file_name = Utility::remove_extension(Parameters::getInstance()->get_filename());
+    std::string tmp = xml_file_name + "_" + file_name + ending;
+    const char *filename = tmp.c_str();
+    Domain *domain = Domain::getInstance();
+    real X1 = domain->get_X1();
+    real Y1 = domain->get_Y1();
+    real Z1 = domain->get_Z1();
+
+    size_t level = field.get_level();
+    int Nx = static_cast<int>(domain->get_Nx(level));
+    int Ny = static_cast<int>(domain->get_Ny(level));
+    int Nz = static_cast<int>(domain->get_Nz(level));
+
+    real dx = domain->get_dx(level);
+    real dy = domain->get_dy(level);
+    real dz = domain->get_dz(level);
+
+    int size = static_cast<int>(domain->get_size());
+
+    // Initialize variables
+    int size_vars = 7; // Number of variables
+    int var_dims[] = {1, 1, 1, 1, 1, 1, 1}; // Dimensions of variables
+    int centering[] = {0, 0, 0, 0, 0, 0, 0}; // Whether the variables are centered in a cell: 0 for zonal!
```

Comment:
two spaces between code and comment are needed.

Reply:
684ed43